client/state/plugins/installed/actions.js

```diff
@@ -51,6 +51,14 @@ import { getSite } from 'calypso/state/sites/selectors';
 
 import 'calypso/state/plugins/init';
 
+// TODO: remove this function when client/state/plugins/installed/test/selectors-ts.ts
+//       replaces client/state/plugins/installed/test/selectors.js
+const isSiteDataOnPluginObject = ( plugin ) => {
```

Comment:
This is here to make these actions work while both the TypeScript and JavaScript versions of the selectors coexist, and it will be removed when the JavaScript selectors are removed. The TypeScript versions place these properties on the site object instead of the plugin object, so by testing for them we can determine which version of the data is being used and handle it appropriately.

Reply:
I changed my mind on this and decided to make it a feature of the `actions.js` file. Sometimes data returned from the server is dispatched directly into one of these actions. Data returned from the server always has the site-specific properties on the plugin object. Because of this I believe it is safest to have the actions able to handle both data structures.

Reply:
Do you think we can address this in a separate PR? So we could keep this substantially smaller and easier to review?

Reply:
I created https://github.com/Automattic/wp-calypso/pull/76684 to change only the actions. We can get that one reviewed and merged first and then rebase this one to remove these changes from this PR. But in the meantime if I take these changes off this PR then this PR won't be able to be tested.